src/Deedle/FrameExtensions.fs

```diff
@@ -285,6 +285,14 @@ type Frame =
     let data = VectorBuilder.Instance.Create(vectors)
     Frame(rowIndex, colIndex, data, IndexBuilder.Instance, VectorBuilder.Instance)
 
+  [<CompilerMessage("This method is not intended for use from F#.", 10001, IsHidden=true, IsError=false)>]
+  static member FromJaggedArray(jArray:'T [][]) =
+    if Array.exists (fun (x:'T []) -> x.Length <> jArray.[0].Length) jArray then
+      invalidOp "FromJaggedArray: The input jagged array must have the same dimensions in all inner arrays."
+    else
+      let arr2D = Array2D.init jArray.Length jArray.[0].Length (fun r c -> jArray.[r].[c])
+      Frame.FromArray2D(arr2D)
```

Comment:
creating a frame like that doesn't look effective . We create a copy of array just to pass it further? We can easily do the same even now

```fsharp
jArray |> array2D |> Frame.FromArray2D
```

Reply:
I was not thinking about performance here, as I see the frame creation as a step that only happens a single time in a data analysis pipeline, but i can definitely adjust thus to work exactly as the FromArray2D function.

>We can easily do the same even now
>jArray |> array2D |> Frame.FromArray2D

Agreed, but the whole point of this PR is to get rid of the function in the middle. And where is the `array2D` from this example coming from anyways? I cant find it in FSharp.Core and needing an external library for that or having to do the copy in the analysis is exactly what i want to prevent.

Reply:
> And where is the array2D from this example coming from anyways 

that's from FSharp.Core

https://github.com/dotnet/fsharp/blob/dbf9a625d3188184ecb787a536ddb85a4ea7a587/src/fsharp/FSharp.Core/fslib-extra-pervasives.fs#L185-L204

Reply:
Well that was a brainfart on my side :D sorry.

Reply:
I used the same code now that is used in FromArray2D so I think it should be now more performant either way